### ISSUE TITLE
Namespaces in attribute values

### ISSUE THREAD

**Issue by finlay** [snoyberg/xml#86]
Hi

I am having trouble with a XML that has namespaced values. For example:
```xml
<Object xmlns:q1="example.org.nz" xsi:type="q1:ObjectType" Other="attributes" are="here" />
```
I would like to compare the value of the `"type"` attribute for equality with
`"{example.org.nz}ObjectType"`, but can't do that. The "q1" is just noise.

See this [MSDN documentation](https://msdn.microsoft.com/en-us/library/system.xml.serialization.xmlserializer.unreferencedobject(v=vs.110).aspx) for where these things come from.

In the meantime I am going to ignore the prefix's in the values, but it is not ideal.

F

**Comment by snoyberg** [comment]
Sounds like #38. Have you seen psRetainNamespaces?

**Comment by finlay** [comment]
Yes, it is very similar to the psRetainNamespaces setting. That handles the name space on the attribute name, but not the attribute value. 

In the example above, we have `xsi:type="q1:ObjectType"`. The `xsi` type is handled properly now, and usefully. But the value `q1:ObjectType` could have the namespace stripped off it, and the `q1` namespace added to the namespaces in the name of the attribute. 
I like the fact that attribute values are currently `Text`, so probably don't want to change that.

Would you be interested in a patch ?

Fin

**Comment by snoyberg** [comment]
Sure, I'd accept a patch to add a new option for resolving namespace values.
